### DISCUSSION TITLE
Switching between Romaji and Kana input methods

### DISCUSSION THREAD

**Discussion by DutchEllie** [google/mozc#891] [Q&A]
On the [Microsoft website](https://learn.microsoft.com/en-us/windows-hardware/design/component-guidelines/keyboard-japan-ime) for Japanese IME they state that the alt+ ImeOn combination should switch between kana input and romaji input methods and it does so in Windows. 

I see in the code for the GUI that there is a [todo](https://github.com/google/mozc/blob/0a8903627b00a4f6b0cea6e6194bd4771ae558b9/src/gui/config_dialog/keybinding_editor.cc#L137) for better supporting the romaji key and I was wondering if this is being worked on.

My apologies, since I am not Japanese I couldn't find if there was an active issue opened right now on this subject. Feel free to link that issue and close mine if it already exists.

**Comment by hiroyuki-komatsu** [comment]
Hi DutchEllie,

We don't have key combinations for switching Kana input (e.g. a→ち) and Romaji input (e.g. a→あ).
We think the users rarely switch Kana input and Romaji input, then the UI configuration is good enough.

That [todo](https://github.com/google/mozc/blob/0a8903627b00a4f6b0cea6e6194bd4771ae558b9/src/gui/config_dialog/keybinding_editor.cc#L137) is about the key labeled Romaji (ローマ字), but not related with this issue.

Thank you,

**Comment by mikunimaru** [comment]
You can change the input method using the following key combinations.
![image](https://hoshino-wp.com/wp-content/uploads/2019/09/123-768x363.jpg)

However, Mozc's default setting disables the switching function, so you need to change the setting and restart Mozc.
![sc](https://user-images.githubusercontent.com/43168745/207822687-879f337c-13b2-4680-9155-6c9dfd772aa9.jpg)

I think it would be useful on a shared PC if this setting could be changed without restarting Mozc.
At least Mozc on Windows seems to require a reboot.

**Comment by hiroyuki-komatsu** [comment]
Hi mikunimaru,

Ah, you are right. I overlooked it. Those two options are available only on Windows,
because Windows has the system level feature to do it.
We have intentionally disabled it by default to avoid switching Kana/Romaji by accident.

By the way, IIRC, you don't need to restart Mozc to switch Kana and Romaji.
Soon after you switch Kana and Romaji on the GUI, it works.

I hope this is a good balance between functionalities and requirements.
Thank you,

**Comment by mikunimaru** [comment]
![a](https://user-images.githubusercontent.com/43168745/211575336-9720ed7a-f447-4bb0-bd71-a3eb8c0d6e3c.jpg)
It looks like switching input methods requires an application restart.
I think it would be convenient for shared PCs if settings can be changed while Mozc is running on Windows.
